apps/api/src/migration/1676405526677-CreateReportView.ts

```diff
@@ -0,0 +1,278 @@
+              asa.status_id IN (
+                SELECT ias.id FROM ien_applicant_status ias WHERE ias.status IN (
+                  'Candidate Passed Interview',
+                  'Candidate Did Not Pass Interview'
+                )
+              )
+          ) as interview,
+          (SELECT min(start_date) FROM public.ien_applicant_status_audit asa
+            WHERE asa.applicant_id = hwa.id AND
+              asa.status_id IN (
+                SELECT ias.id FROM ien_applicant_status ias WHERE ias.status IN (
+                  'Candidate Passed Reference Check',
+                  'Candidate Did Not Pass Reference Check'
+                )
+              )
+          ) as reference_check,
+          (SELECT min(start_date) FROM public.ien_applicant_status_audit asa
+            WHERE asa.applicant_id = hwa.id AND
+              asa.status_id IN (
+                SELECT ias.id FROM ien_applicant_status ias WHERE ias.status IN (
+                  'Job Offer Accepted',
+                  'Job Offer Not Accepted',
+                  'Job Competition Cancelled',
+                  'HA is Not Interested',
+                  'No Position Available',
+                  'Candidate Withdrew from Competition'
+                )
+              )
+          ) as competition_outcome,
+          (SELECT min(start_date) FROM public.ien_applicant_status_audit asa
+            WHERE asa.applicant_id = hwa.id AND
+              asa.status_id IN (
+                SELECT ias.id FROM ien_applicant_status ias WHERE ias.category = 'BC PNP Process'
+              )
+          ) as immigration,
+          (SELECT min(start_date) FROM public.ien_applicant_status_audit asa
+            WHERE asa.applicant_id = hwa.id AND
+              asa.status_id = (SELECT ias.id FROM ien_applicant_status ias WHERE ias.status = 'Sent First Steps document to candidate')   
+          ) as sent_first_steps_document,
+          (SELECT min(start_date) FROM public.ien_applicant_status_audit asa
+            WHERE asa.applicant_id = hwa.id AND
+              asa.status_id = (SELECT ias.id FROM ien_applicant_status ias WHERE ias.status = 'Sent employer documents to HMBC')   
+          ) as sent_employer_documents_to_hmbc,
+          (SELECT min(start_date) FROM public.ien_applicant_status_audit asa
+            WHERE asa.applicant_id = hwa.id AND
+              asa.status_id = (SELECT ias.id FROM ien_applicant_status ias WHERE ias.status = 'Submitted application to BC PNP')   
+          ) as submitted_bc_pnp_application,
+          (SELECT min(start_date) FROM public.ien_applicant_status_audit asa
+            WHERE asa.applicant_id = hwa.id AND
+              asa.status_id = (SELECT ias.id FROM ien_applicant_status ias WHERE ias.status = 'Received Confirmation of Nomination')   
+          ) as received_confirmation_of_nomination,
+          (SELECT min(start_date) FROM public.ien_applicant_status_audit asa
+            WHERE asa.applicant_id = hwa.id AND
```

Comment:
Lots of repetition here. Would it be possible to create a list constant with status and the variable name as an object, and then loop through it? 
e.g const status_list = [ {status : 'Sent First steps document to candidate', name:'sent_first_steps_document'}]